File: src/1046.java
import java.util.Collections;
import java.util.PriorityQueue;
import java.util.Queue;

class Solution {
    public int lastStoneWeight(int[] stones) {
        Queue<Integer> q = new PriorityQueue<>(stones.length, Collections.reverseOrder());

        for (int i = 0; i < stones.length; i++) {
            q.offer(stones[i]);
        }

        while (q.size() > 1) {
            int y = q.poll();
            int x = q.poll();

            if (x != y) { q.offer(y - x); }
        }

        return q.isEmpty() ? 0 : q.poll();
    }
}

File: src/main/javascript/1046.js
/**
 * @param {number[]} stones
 * @return {number}
 */
var lastStoneWeight = function(stones) {
    var x;
    var y;

    for(var i = 0; stones.length > 1; i++) {
        stones.sort((a, b) => a > b);

        y = stones.pop();
        x = stones.pop();

        if(x !== y) {
            y -= x;
            stones.push(y);
        }
    }

    return stones[0] || 0;
};
